Development v1

Andretalik opened the pull request:
Final commit for presentation

Andretalik committed:
Added the Classes for Dojo, Room & Person with tests

Andretalik committed:
Added correct Classes and respective tests

Andretalik committed:
Added tests and functionality for assigning rooms to created people

Andretalik committed:
Added functionality for random allocation of rooms to people with respective tests

Andretalik committed:
Added functionality for maximum capacity of rooms

Andretalik committed:
Added functionality to print all occupants in a room and its respective test

Andretalik committed:
Removal of commented code and creation of docopt_main.py

Andretalik committed:
Removed unneccessary variable name to save on space

Andretalik committed:
Added super __init__ calls and functionality to ensure allocated room has space

Andretalik committed:
Added return statemant for use in docopts implementation

Andretalik committed:
Deletion of redundant code to adhere to D.R.Y.

Andretalik committed:
Added tests wrong syntax(sad tests).

Andretalik committed:
Deletion of uncompleted code

Andretalik committed:
Creation of functioning docopt with tweaks in functions to accommodate it

Andretalik committed:
Formatted output from function to match output expected

alex-magana commented on Dojo_class.py:
```diff
@@ -0,0 +1,121 @@
+
+class Dojo(object):
+    def __init__(self):
+        self.all_rooms = []
```
The all_rooms variable can store room attributes. This should eliminate the need for all_offices and all_living_spaces

alex-magana commented on Dojo_class.py:
```diff
@@ -0,0 +1,121 @@
+        self.all_rooms = []
+        self.livingspaces = []
+        self.offices = []
+        self.all_people = []
```
Apply the concept indicated above to all_people, fellows and staff

alex-magana commented on Dojo_class.py:
```diff
@@ -0,0 +1,121 @@
+            for room_name in rooms_to_make:
+                if room_type == "office":
+                    office_instance = Office("office", room_name)           # to ensure the instance being worked on
+                    self.all_rooms.append(office_instance)                  # during the function call is singular
```
Consider use of a function that does assignment of rooms. Principle of single responsibility.

alex-magana commented on Dojo_class.py:
```diff
@@ -0,0 +1,121 @@
+                    self.all_rooms.append(livingspace_instance)
+                    self.livingspaces.append(livingspace_instance)
+                    self.available_livingspaces.append(livingspace_instance)
+                    print("A {} called {} has been created".format(livingspace_instance.room_type, livingspace_instance.room_name))
```
Limit length of lines to 72 characters.
https://www.python.org/dev/peps/pep-0008/

kimobrian reviewed (commented):
+1 for a .gitignore
A part from redundancy and code optimization , you really understand your code base. 
+1 for branching your repo.
-Learn about setup and teardown in testing.
-Learn more on data structures(lists, dictionaries etc) esp list comprehensions and control structures(if, while etc)
-Learn on pep8 style of writing code.

Andretalik committed:
Adding badge url to README and .travis.yml

Andretalik committed:
Added config to the .travis.yml

Andretalik committed:
Attmepted fix to the .travis.yml

Andretalik committed:
Fix: Commented out coveralls since unintergrated

Andretalik committed:
Fix script error in Travis

Andretalik committed:
Fixed the badge to show the correct value

Andretalik committed:
Added notifications for Slack

Andretalik committed:
Added the pylama to requirement.txt

Andretalik committed:
Attempted coveralls fix

Andretalik committed:
Added coveralls badge

Andretalik committed:
Attempted fix for coveralls script

Andretalik committed:
Attempted fix for coveralls script in .travis.yml

Andretalik committed:
Attempted fix for coveralls script in .travis.yml

Andretalik committed:
Added the coveralls to the Venv and requirements.txt

Andretalik committed:
Removed echo foo command

Andretalik committed:
Added .coveralls.yml to branch

Andretalik committed:
Fixed pep8 issues

Andretalik committed:
Attemped fix in coveralls.yml

Andretalik committed:
Renamed test file and ignored .pyc

Andretalik committed:
Fix for coverage

Andretalik committed:
Fix for broken coverage

Andretalik committed:
Added coveralls to .travis.yml

Andretalik committed:
Attempted fix for coveralls

Andretalik committed:
Added .coveragerc

Andretalik committed:
Removed coveralls badge from README.md

Andretalik committed:
Added Coveralls badge to README.md

Andretalik committed:
Added .DS_Store to .gitignore

Andretalik committed:
Changed coveralls badge from svg. to png.

Andretalik committed:
Changed Original markdown to sheilds.io format

Andretalik committed:
Added .coverage to .gitignore

Andretalik committed:
Fixed the import all from Room and Person

Andretalik committed:
Fixed linter errors

Andretalik committed:
Added Title and some content

Andretalik committed:
Removed private .coverage file

Andretalik committed:
Removed some unneccessary comments and updated the README.md

Andretalik committed:
Removed uneccessary lists

- Removing the lists that are redundant and taking up space for nothing.

Andretalik committed:
Fixed parameters for the Person class

 - Added an ID for each person.
 - Added a second name parameter for each person.

Andretalik committed:
Refactoring the codebase

 - Added the second name parameter for the Person Class.
 - Removed misplaced code that printed the person.type.

Andretalik committed:
Added Tests

 - Added tests for printing allocations
 - Added tests for printing unallocated people
 - Added tests for reallocations people

Andretalik committed:
Renamed the files to conform to the Python naming Conventions

Andretalik committed:
Removed uneccessary variable for creating a text file`

Andretalik committed:
Added Tests

 - Added tests for the print_allocation function
 - Added tests for the print_unallocated function
 - Fixed a test: TestMembersOfRooms print_room command

Andretalik committed:
Added tests for loading people from a file

Andretalik committed:
Added tests

 - Added missing teardowns
 - Added test for deletion of people

Andretalik committed:
Added tests

 - Added test for deletion of rooms

Andretalik committed:
Added New Tests

 - Added the basic tests for the databases

Andretalik committed:
Added tests

 - Added tests for the load_db function

Andretalik committed:
Added and fixed tests

 - Fixed the test_delete_person
 - Added test_load_correct_file

Andretalik committed:
Fix the import error in test_accommodation.py

Andretalik committed:
Fixed the failures in tests

 - Amended the tests to reflect the changes in the methods

Andretalik committed:
Fixed the formatting errors in outputs

 - Added the access point to show the room names rather than just the object itself

Andretalik committed:
Updated the badges for the specific branch

Andretalik committed:
Fixed the refactoring problems in the tests

 - The addition of a second name for people incorporated into the tests
 - The change in the manner in which the parameters for offices are passed in

Andretalik committed:
Fix the changes in attributes caused by refactoring

Andretalik committed:
Fixes broken syntax in add_person method

 - Fixes method to return strings rather than print them out

Andretalik committed:
Removed the occcurences nonexistent attributes

 - Removed occurences of self.livingspaces and self.offices

Andretalik committed:
Removed commented out code in travis.yml

Andretalik committed:
Fixed syntax error killing test code

Andretalik committed:
Removed the commented out code.

Andretalik committed:
Added more information concerning the application

Andretalik committed:
Fixed the docstring and added another method

 - Added command for print_allocations
 - Updated docstring to reflect the above addition

Andretalik committed:
Added function to close file after it was opened

Andretalik committed:
Added *.txt files to be untracked

Andretalik committed:
Added docstrings to all methods within the controller

Andretalik committed:
Fixed test_print_unallocated

 - Fixed the test code where it was testing a file that doesn't exist

Andretalik committed:
Restructured the print_allocations function

 - Used more pythonic ways of manipulating data
 - Writes all data to the file in one go rather than line by line

Andretalik committed:
Fix the issues with my functions not calling each other

 - Fix the add_person function not calling the allocations functions

Andretalik committed:
Add print_unallocated function to Docopt

Andretalik committed:
Fix super() function in class

 - Make each class only initialise the parameters it will use.

Andretalik committed:
Clean up code

 - Removed commented out code

Andretalik committed:
Fix the super() function within the classes LivingSpace and Office.

Andretalik committed:
Fix the test_add_person_fellow_successful

 - Change the variable name to be explicit

Andretalik committed:
Add docstrings for functions

Andretalik committed:
Fix the 'None' print bug

Andretalik committed:
Add the close() function after opening the files

Andretalik committed:
Removed incomplete code

Andretalik committed:
Merge pull request #3 from Andretalik/test_full_project

Test full project
